### DISCUSSION TITLE
Heat map "IR Drop" has not been populated with data

### DISCUSSION THREAD

**Discussion by b224hisl** [The-OpenROAD-Project/OpenROAD-flow-scripts#2047] [Q&A] [answered]
Hello,
After I finish my work, I execute `read_db 6_final.odb`.  Then I can see the die view of the chip. But When I click `Heat Maps -> IR Drop`. It reports "Heat map "IR Drop" has not been populated with data".
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/83a57b95-86b7-4e2c-a5aa-68d17acc5b39)

I think I have generated all the files (gds,odb,spef,,,,) sucessfully. Do you know how to handle this problem? Thank you very much~

**Comment by maliberty** [comment]
You need to run analyze_power_grid as the results are not stored in the DB

**Reply by b224hisl** [reply]
how to load the DB with `analyze_power_grid`。 Do you mean that I should execute that command and then save the db and load it ?

**Reply by rovinski** [reply, marked as answer]
Load the DB, then run `analyze_power_grid`. Heatmap results are temporary and aren't saved once you exit OpenROAD.

**Reply by b224hisl** [reply]
Thank you very much! Similarly, `report_worst_slack/report_wns/report_cts` can not take effect when just loading the db. what should I execute before?

**Reply by maliberty** [reply]
It easier to use ORFS' `make open_<step>` or `make gui_<step>` which will load the necessary files.  For example `make gui_final`

**Reply by b224hisl** [reply]
> You need to run analyze_power_grid as the results are not stored in the DB

I have re-installed the OR to the latest version.  Because the heatmap results are temporary and aren't saved once I exit OpenROAD, how to load the IR drop heat map again?

I have tried ‘analyze_power_grid -net VDD/VSS’ as you previously suggested, but it doesn't work, it will get meaningless results:
![QQ_1729942490553](https://github.com/user-attachments/assets/2071ed0f-9f13-4a81-8b64-d44934791e8e)

Here is the executing log of this design:
![QQ_1729942560868](https://github.com/user-attachments/assets/a087f726-3214-48c3-b3d2-29bf1d0d3403)

So I think there might be a way to reload the IR drop heatmap and analysis results after I exit OR.

**Comment by b224hisl** [comment]
I haven't used OpenROAD Flow Scripts (ORFS) for some time, and there have been quite a few updates. I am still encountering the issue 'Heat map "IR Drop" has not been populated with data.' The previous methods no longer work. What is the current method to load the IR drop heatmap in the GUI? It seems that the` save_images.tcl` has this problem:
![image](https://github.com/user-attachments/assets/b28f4aea-4288-44b2-9975-653931068531)

**Reply by b224hisl** [reply]
For this problem, I have pulled a new issue https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/issues/2510

**Comment by maliberty** [comment]
Not much has changed and we generate images nightly in the CI.  I suspect you have an out of date OR app and need to update.
